flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -179,37 +186,93 @@ public Builder writeParallelism(int newWriteParallelism) {
         try (TableLoader loader = tableLoader) {
           this.table = loader.loadTable();
         } catch (IOException e) {
-          throw new UncheckedIOException("Failed to load iceberg table from table loader: " + tableLoader, e);
+          throw new UncheckedIOException(
+              "Failed to load iceberg table from table loader: " + tableLoader,
+              e);
         }
       }
 
       IcebergStreamWriter<RowData> streamWriter = createStreamWriter(table, tableSchema);
       IcebergFilesCommitter filesCommitter = new IcebergFilesCommitter(tableLoader, overwrite);
 
-      this.writeParallelism = writeParallelism == null ? rowDataInput.getParallelism() : writeParallelism;
-
-      DataStream<Void> returnStream = rowDataInput
-          .transform(ICEBERG_STREAM_WRITER_NAME, TypeInformation.of(DataFile.class), streamWriter)
-          .setParallelism(writeParallelism)
-          .transform(ICEBERG_FILES_COMMITTER_NAME, Types.VOID, filesCommitter)
+      DataStream<?> returnStream = rowDataInput
+          .transform(ICEBERG_STREAM_WRITER_NAME,
+              TypeInformation.of(DataFile.class), streamWriter)
+          .setParallelism(rowDataInput.getParallelism())
```

Comment:
Why change this ? we will need to allow users to set their own writer parallelism, so I think we may need to revert this ?